### PR TITLE
Fixes #14833: Use rudder agent check at postinst and factory reset to avoid duplacing code

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/14833

### CODE REVIEW

File: share/commands/agent-check [hunk 1]
```diff
@@ -60,24 +64,11 @@ init_commands
 MYUID=`id | cut -d\( -f2 | cut -d\) -f1`
 if [ "${MYUID}" != 'root' ];then echo "You must be root"; exit; fi
 
-# Get the value of rudder-agent run interval from file /var/rudder/cfengine-community/inputs/run_interval
-if [ -f "${CFE_DIR}/inputs/run_interval" ]; then
-  RUN_INTERVAL=`cat "${CFE_DIR}/inputs/run_interval"`
-  # If the value is not a number, reset to 5
-  if ! test "${RUN_INTERVAL}" -gt 0 2>/dev/null
```
Comment (amousset) [inline share/commands/agent-check]: shouldn't we move this default value to common.sh to handle missing rudder.json file (i.e. before bootstraping fir example)

File: share/commands/agent-check [hunk 2]
```diff
@@ -221,20 +235,34 @@ check_and_fix_rudder_uuid() {
...
+  if [ ! -s "${CFE_DIR}/inputs/common/1.0/update.cf" ] || [ ! -s "${CFE_DIR}/inputs/failsafe.cf" ] || [ ! -s "${CFE_DIR}/inputs/promises.cf" ]
   then
-    printf "ERROR: There was an error during promises update. Reseting to initial promises and updating..."
-    rm -rf ${CFE_DIR}/inputs/*
-    if [ -d "/opt/rudder/share/initial-promises" ] # rudder package has been properly installed
-    then
-      rudder agent reset
-    fi
-    echo " Done"
+    [ "$QUIET" = false ] && printf "INFO: Promises absent restoring initial version, and updating ..."
```
Comment (amousset) [inline share/commands/agent-check]: "Policies absent, restoring"...

File: share/commands/agent-check [hunk 3]
```diff
@@ -221,20 +235,34 @@ check_and_fix_rudder_uuid() {
...
+    rudder agent reset ${OPT}
+    [ "$QUIET" = false ] && echo " Done"
   fi
+  if ! ${CFE_DIR}/bin/cf-promises -f failsafe.cf > /dev/null || ! ${CFE_DIR}/bin/cf-promises > /dev/null
+  then
+    [ "$QUIET" = false ] && printf "${YELLOW}WARNING${NORMAL}: Promises invalid, reseting to initial promises and updating..."
```
Comment (amousset) [inline share/commands/agent-check]: Promises -> Policies

**Comment by peckpeck** [comment]
Commit modified

File: share/commands/agent-check [hunk 4]
```diff
@@ -113,9 +127,9 @@ check_and_fix_cfengine_processes() {
   CF_EXECD_RUNNING=`${PS_COMMAND} | grep "${CFE_DIR}/bin/cf-execd" | sed -e '/grep/d' | cat`
   NB_CF_EXECD_RUNNING=`echo "${CF_EXECD_RUNNING}" | sed -e '/^$/d' | wc -l`
   if [ ${NB_CF_EXECD_RUNNING} -gt 1 ]; then
-    printf "WARNING: Too many instance of CFEngine cf-execd processes running. Killing them..."
+    [ "$QUIET" = false ] && printf "${YELLOW}WARNING${NORMAL}: Too many instance of CFEngine cf-execd processes running. Killing them..."
```
Comment (amousset) [inline share/commands/agent-check]: CFEngine cf-execd -> Rudder cf-execd

File: share/commands/agent-check [hunk 5]
```diff
@@ -126,28 +140,28 @@ check_and_fix_cfengine_processes() {
...
+    rudder agent start ${OPT}
+    [ "$QUIET" = false ] && echo " Done"
   fi
 
   # Check for anomalous number of CFEngine processes
   # If there are more than 6 agent/executor processes, we should kill them, and purge the lock database
   # 6 means 1 cf-execd + 5 cf-agent
   if [ "${NB_CF_PROCESS_RUNNING}" -gt 6 ]; then
-    printf "WARNING: Too many instance of CFEngine processes running. Killing them, and purging the CFEngine lock database..."
+    [ "$QUIET" = false ] && printf "${YELLOW}WARNING${NORMAL}: Too many instance of CFEngine processes running. Killing them, and purging the CFEngine lock database..."
```
Comment (amousset) [inline share/commands/agent-check]: CFEngine processes -> Agent processes

Comment (amousset) [reply share/commands/agent-check]: CFengine lock -> agent lock

File: share/commands/agent-check [hunk 6]
```diff
@@ -103,6 +101,22 @@ check_and_fix_cfengine_keys() {
...
 }
 
+# There must be a policy server otherwise we can't do anything
+check_policy_server_or_exit() {
+  if [ ! -s ${CFE_DIR}/policy_server.dat ]
+  then
+    echo "********************************************************************************"
+    echo "rudder-agent has been installed (not started). This host can be a Rudder node."
+    echo "To get started, configure your Rudder server's hostname and continue the process:"
+    echo "# echo 'rudder.server' > ${CFE_DIR}/policy_server.dat"
```
Comment (amousset) [inline share/commands/agent-check]: we should display /opt/rudder/etc instead here

**Comment by peckpeck** [comment]
Commit modified

File: share/commands/agent-check (hunk 6 above)
Comment (peckpeck) [reply share/commands/agent-check]: I think this is the lase place that should be changed since if the symlink was not created this will fail

**Pull request merged**